### PR TITLE
Internal: Merge experiment feature to 3.8 [ED-7714]

### PR DESCRIPTION
**Pull request by TzviRabinovitch**

### CODE REVIEW

File: includes/admin-templates/new-template.php [hunk 1]
```diff
@@ -83,7 +78,13 @@
 		 *
 		 * @since 2.0.0
 		 */
-		do_action( 'elementor/template-library/create_new_dialog_fields' );
+		do_action( 'elementor/template-library/create_new_dialog_fields', $new_template_control_form );
+
+		$additional_controls = $new_template_control_form->get_controls();
+		if ( $additional_controls ) {
+			wp_add_inline_script( 'elementor-admin', 'var elementor_new_template_form_controls = ' . wp_json_encode( $additional_controls ) . ';' );
```
Comment (StyleShit) [inline includes/admin-templates/new-template.php:85]:
```suggestion
			wp_add_inline_script( 'elementor-admin', 'const elementor_new_template_form_controls = ' . wp_json_encode( $additional_controls ) . ';' );
```

File: includes/template-library/forms/new-template-form.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	 * @throws \Exception
+	 */
+	public function render() {
+		foreach ( $this->get_controls() as $control ) {
+			switch ( $control['type'] ) {
+				case Controls_Manager::SELECT:
+					$this->render_select( $control );
+					break;
+				default:
+					throw new \Exception( 'Not supported control type' );
```
Comment (StyleShit) [inline includes/template-library/forms/new-template-form.php:27]:
```suggestion
					throw new \Exception( '"' . $control['type'] . '" control type is not supported );
```

Maybe?

File: includes/template-library/forms/new-template-form.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+			}
+		}
+	}
+
+	private function render_select( $control_settings ) {
+		$control_id = "elementor-new-template__form__{$control_settings['name']}";
+		$wrapper_class = isset( $control_settings['wrapper_class'] ) ? $control_settings['wrapper_class'] : '';
+		?>
+		<div id="<?php echo esc_html( $control_id ); ?>__wrapper" class="elementor-form-field <?php echo esc_html( $wrapper_class ); ?>">
```
Comment (StyleShit) [inline includes/template-library/forms/new-template-form.php:36]:
```suggestion
		<div id="<?php echo esc_attr( $control_id ); ?>__wrapper" class="elementor-form-field <?php echo esc_attr( $wrapper_class ); ?>">
```

also others

**Review by StyleShit — CHANGES_REQUESTED** [review]

File: assets/dev/js/admin/new-template/layout.js [hunk 4]
```diff
@@ -17,9 +17,25 @@ module.exports = elementorModules.common.views.modal.Layout.extend( {
...
 		this.showLogo();
 
 		this.showContentView();
+
+		const dynamicFieldsVisibilityListener = () => {
+			elementorAdmin.templateControls.setDynamicFieldsVisibility( lookupControlIdPrefix, elementor_new_template_form_controls );
+		};
+
+		this.getModal().onShow = () => {
+			elementorAdmin.templateControls.setDynamicFieldsVisibility( lookupControlIdPrefix, elementor_new_template_form_controls );
```
Comment (matipojo) [inline assets/dev/js/admin/new-template/layout.js:32]:
```suggestion
			dynamicFieldsVisibilityListener();
```

File: assets/dev/js/admin/new-template/template-controls.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
+export default class TemplateControls {
+	setDynamicFieldsVisibility( lookupControlIdPrefix, controls ) {
```
Comment (matipojo) [inline assets/dev/js/admin/new-template/template-controls.js:2]:
```suggestion
	setDynamicControlsVisibility( lookupControlIdPrefix, controls ) {
```

File: core/admin/admin.php [hunk 6]
```diff
@@ -574,7 +574,9 @@ public function admin_action_new_post() {
 		 * @param array $meta Post meta data.
 		 */
 		$meta = apply_filters( 'elementor/admin/create_new_post/meta', $meta );
-
+		if ( isset( $_GET['meta'] ) && is_array( $_GET['meta'] ) ) {
+			$meta = array_merge( array_map( 'sanitize_text_field', $_GET['meta'] ), $meta );
+		}
```
Comment (matipojo) [inline core/admin/admin.php:579]: Insert before the filter

**Review by matipojo — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/components/documents/commands/internal/attach-preview.js [hunk 7]
```diff
@@ -34,7 +34,7 @@ export class AttachPreview extends $e.modules.CommandInternalBase {
 	}
 
 	attachDocumentToPreview( document, args ) {
-		const { selector = '.elementor-' + document.id } = args;
+		const { selector = '.elementor-' + document.id, scroll = true } = args;
```
Comment (StyleShit) [inline assets/dev/js/editor/components/documents/commands/internal/attach-preview.js:37]:
```suggestion
		const { selector = '.elementor-' + document.id, shouldScroll = true } = args;
```
Maybe?

File: core/base/document.php [hunk 8]
```diff
@@ -320,7 +320,8 @@ public function get_container_attributes() {
 			}
 		}
 
-		return $attributes;
+		// apply this filter to allow the attributes to be modified by different sources
```
Comment (StyleShit) [inline core/base/document.php:323]: kinda redundant

**Review by StyleShit — CHANGES_REQUESTED** [review]

File: core/admin/admin.php (hunk 6 above)
Comment (TzviRabinovitch) [reply core/admin/admin.php:579]: done

File: assets/dev/js/editor/components/documents/commands/internal/attach-preview.js (hunk 7 above)
Comment (TzviRabinovitch) [reply assets/dev/js/editor/components/documents/commands/internal/attach-preview.js:37]: Done

File: assets/dev/js/admin/new-template/template-controls.js (hunk 5 above)
Comment (TzviRabinovitch) [reply assets/dev/js/admin/new-template/template-controls.js:2]: Done

File: assets/dev/js/admin/new-template/layout.js (hunk 4 above)
Comment (TzviRabinovitch) [reply assets/dev/js/admin/new-template/layout.js:38]: Done

**Pull request merged**